### PR TITLE
refactor opentelemetry_exporter

### PR DESCRIPTION
**Pull request by SergeTupchiy**
- use grpc-erl client instead of grpcbox
- support exporter timeout
- properly handle per-signal configuration
- give precedence to directly passed exporter options over app env/ENV vars

Motivation for chaning grpc client: 
- grpcbox latest release (0.16.0) has hardcoded receive timeout of 5s: https://github.com/tsloughter/grpcbox/blob/v0.16.0/src/grpcbox_client.erl#L81
- the main branch has fixes for that, so I asked the maintaner to create a new release
- Meanwhile, I started testing the main branch of grpcbox and ran into an issue with sending large payloads, it was stuck here: https://github.com/tsloughter/chatterbox/blob/master/src/h2_stream_set.erl#L1042
The request was just hanging and timing out...
 I'm yet to report this issue to the upstream repo, but I decided that grpc-erl is ready to use and should be more resilient overall..

### CODE REVIEW

File: apps/opentelemetry_exporter/rebar.config [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {erl_opts, [debug_info]}.
-{deps, [{grpcbox, ">= 0.0.0"},
-        {tls_certificate_check, "~> 1.18"},
+
+{deps, [{grpc, {git, "https://github.com/emqx/grpc-erl", {tag, "0.6.11"}}},
```
Comment (zmstone) [inline apps/opentelemetry_exporter/rebar.config:3]: could you maybe add some words why this change in the PR description?

File: rebar.config [hunk 2]
```diff
@@ -61,12 +59,12 @@
                 opentelemetry_zipkin_pb]}.
 
 
-{dialyzer, [{warnings, [no_unknown]}]}.
+%{dialyzer, [{warnings, [no_unknown]}]}.
```
Comment (zmstone) [inline rebar.config:62]: Q: is it a todo ?

File: apps/opentelemetry_exporter/src/opentelemetry_exporter.erl [hunk 3]
```diff
@@ -359,18 +262,155 @@ export(_, _Tab, _Resource, _State) ->
...
+    end;
+init_by_proto(HTTP, State) when HTTP =:= http_protobuf; HTTP =:= http_json ->
+    HttpcProfile = start_httpc(State),
+    {ok, State#state{httpc_profile=HttpcProfile}}.
 
-grpcbox_endpoints(Endpoints) ->
-    [{scheme(Scheme), Host, Port, maps:get(ssl_options, Endpoint, [])} ||
-        #{scheme := Scheme, host := Host, port := Port} = Endpoint <- Endpoints].
+%% Only called on a parsed and recomposed URL which is guaranteed to be a string
+is_ssl([$h,$t,$t,$p,$s,$:,$/,$/ | _]) -> true;
```
Comment (zmstone) [inline apps/opentelemetry_exporter/src/opentelemetry_exporter.erl]: nit;

`you can do is_ssl("https://" ++ _) -> true;`

File: rebar.config (hunk 2 above)
Comment (SergeTupchiy) [reply rebar.config:62]: Not sure, but this conf causes dialyzer errors:
```
===> Error in dialyzing apps: Unknown dialyzer warning option: no_unknown
```
I commented it out and dialyzer worked fine...

**Review by zmstone — APPROVED** [review]

File: apps/opentelemetry_exporter/rebar.config (hunk 1 above)
Comment (SergeTupchiy) [reply apps/opentelemetry_exporter/rebar.config:3]: Done

File: apps/opentelemetry_exporter/src/opentelemetry_exporter.erl (hunk 3 above)
Comment (SergeTupchiy) [reply apps/opentelemetry_exporter/src/opentelemetry_exporter.erl]: fixed, thanks
